File: backend/src/main/java/com/campusmarketplace/Entity/Post.java
package com.campusmarketplace.Entity;

import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String description;
    private Double askingPrice;
    private String category;
    private String imageUrl;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private User Owner;
}

File: Frontend/campus-marketplace/src/types/post.ts
export interface User {
  id: number;
  username: string;
  email: string;
  college: string;
}

export interface Post {
  id: number;
  title: string;
  description: string;
  askingPrice: number;
  category: string;
  imageUrl: string;
  Owner: User;
}